packages/fleather/lib/src/widgets/text_selection.dart

```diff
@@ -660,176 +879,777 @@ class EditorTextSelectionGestureDetectorBuilder {
   ///
   /// See also:
   ///
-  ///  * [EditorTextSelectionGestureDetector.onForcePressEnd], which triggers this
+  ///  * [TextSelectionGestureDetector.onForcePressEnd], which triggers this
   ///    callback.
   @protected
   void onForcePressEnd(ForcePressDetails details) {
     assert(delegate.forcePressEnabled);
-    renderEditor!.selectWordsInRange(
+    renderEditor.selectWordsInRange(
       from: details.globalPosition,
       cause: SelectionChangedCause.forcePress,
     );
-    if (shouldShowSelectionToolbar) editor!.showToolbar();
+    if (shouldShowSelectionToolbar) {
+      editor.showToolbar();
+    }
   }
 
-  /// Handler for [EditorTextSelectionGestureDetector.onSingleTapUp].
+  /// Handler for [TextSelectionGestureDetector.onSingleTapUp].
   ///
   /// By default, it selects word edge if selection is enabled.
   ///
   /// See also:
   ///
-  ///  * [EditorTextSelectionGestureDetector.onSingleTapUp], which triggers
+  ///  * [TextSelectionGestureDetector.onSingleTapUp], which triggers
   ///    this callback.
   @protected
-  void onSingleTapUp(TapUpDetails details) {
+  void onSingleTapUp(TapDragUpDetails details) {
     if (delegate.selectionEnabled) {
-      renderEditor!.selectWordEdge(cause: SelectionChangedCause.tap);
+      // It is impossible to extend the selection when the shift key is pressed, if the
+      // renderEditor.selection is invalid.
+      final bool isShiftPressedValid = _isShiftPressed;
+      switch (defaultTargetPlatform) {
+        case TargetPlatform.linux:
+        case TargetPlatform.macOS:
+        case TargetPlatform.windows:
+          break;
+        // On desktop platforms the selection is set on tap down.
+        case TargetPlatform.android:
+          if (isShiftPressedValid) {
+            _extendSelection(details.globalPosition, SelectionChangedCause.tap);
+            return;
+          }
+          renderEditor.selectPosition(cause: SelectionChangedCause.tap);
+        //TODO: show spell check suggestions
+        case TargetPlatform.fuchsia:
+          if (isShiftPressedValid) {
+            _extendSelection(details.globalPosition, SelectionChangedCause.tap);
+            return;
+          }
+          renderEditor.selectPosition(cause: SelectionChangedCause.tap);
+        case TargetPlatform.iOS:
+          if (isShiftPressedValid) {
+            // On iOS, a shift-tapped unfocused field expands from 0, not from
+            // the previous selection.
+            final TextSelection? fromSelection = renderEditor.hasFocus
+                ? null
+                : const TextSelection.collapsed(offset: 0);
+            _expandSelection(
+              details.globalPosition,
+              SelectionChangedCause.tap,
+              fromSelection,
+            );
+            return;
+          }
```

Comment:
Guessing this is as implemented in Flutter right?

Reply:
Yes. You check here: https://github.com/flutter/flutter/blob/db7ef5bf9f59442b0e200a90587e8fa5e0c6336a/packages/flutter/lib/src/widgets/text_selection.dart#L2317